### PR TITLE
Add crd-match-check

### PR DESCRIPTION
**Pull request by ivyostosh**
*Issue #, if available:* https://github.com/aws/eks-anywhere-packages/issues/365

*Description of changes:* Add crd match check to presubmit

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by lewisdiamond** [comment]
Should we not just make helm/build be run by default instead?

**Comment by ivyostosh** [comment]
After discussion, we decided to use `git` to detect diffs in `presubmit` directly. 

Note we considered `git diff-index --quiet HEAD --` but passed on it as it will detect timestamp differences as diff too.

File: Makefile [hunk 1]
```diff
@@ -84,6 +84,21 @@ $(GOBIN)/setup-envtest: ## Install setup-envtest
 	# https://github.com/kubernetes-sigs/kubebuilder/issues/2480 
 	go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
 
+TMP_CONFIG_CRD_DIR=config/crd/tmp
+TMP_HELM_CRD_DIR=charts/eks-anywhere-packages/crds/tmp
+crd-match-check: controller-gen # Check all CRD files are properly updated
```
[the remark this answers is not in the document]
Comment (ivyostosh) [reply Makefile]: Updated. I only excluded the Makefile from git diff detection now, not sure if there will be other false positives.

**Review by TerryHowe — APPROVED** [review]
/lgtm
/approve

**Review by TerryHowe — APPROVED** [review]
/lgtm
/approve

**Pull request merged**